temporian/core/operators/rename.py

```diff
@@ -0,0 +1,214 @@
+                        f" {event_names}."
+                    )
+
+            return rename_dict
+
+        super().__init__()
+
+        self.features = {}
+
+        if isinstance(features, str):
+            # check that event only has one feature
+            if len(event.features) != 1:
+                raise ValueError(
+                    "Expected event to have only one feature when passed a"
+                    f" single string. Got {len(event.features)} features"
+                    " instead."
+                )
+            if not features:
+                raise ValueError("Expected feature to be a non-empty string.")
+            only_feature_name = event.features[0].name
+            self.features = {only_feature_name: features}
+
+        elif isinstance(features, dict):
+            # check that every key is a feature name in event
+            feature_names = [feature.name for feature in event.features]
+            self.features = check_rename_dict(features, feature_names)
+
+        event_index_names = event.index_names
+
+        self.index = {}
+
+        if isinstance(index, str):
+            # check that event only has one index
+            if len(event_index_names) != 1:
+                raise ValueError(
+                    "Expected event to have only one index when passed a"
+                    " single string in index. Got"
+                    f" {len(event_index_names)} indexes instead."
+                )
+            # check index is non empty
+            if not index:
+                raise ValueError("Expected index to be a non-empty string.")
+            only_index_name = event_index_names[0]
+            self.index = {only_index_name: index}
+
+        elif isinstance(index, dict):
+            # check that every key is an index name in event
+            self.index = check_rename_dict(index, event_index_names, "index")
+
+        self.add_attribute("features", self.features)
+
+        self.add_attribute("index", self.index)
+
+        # inputs
+        self.add_input("event", event)
+
+        output_sampling = event.sampling
+
+        if index:
+            output_sampling = self.new_sampling(event.sampling)
+
+        # outputs
+        output_features = [  # pylint: disable=g-complex-comprehension
+            Feature(
+                name=self.new_feature_name(f),
+                dtype=f.dtype,
+                sampling=output_sampling,
+                creator=self,
+            )
+            for f in event.features
+        ]
+
+        self.add_output(
+            "event",
+            Event(
+                features=output_features,
+                sampling=output_sampling,
+                creator=self,
+            ),
+        )
+
+        self.check()
+
+    def new_feature_name(self, feature: Feature) -> str:
+        if feature.name in self.features:
+            return self.features[feature.name]
+
+        return feature.name
```

Comment:
nit: This function could be removed and use `self.features.get(feature.name, feature.name)` instead.